llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1027,6 +1027,10 @@ InstructionCost VPWidenIntrinsicRecipe::computeCost(ElementCount VF,
   for (const auto &[Idx, Op] : enumerate(operands())) {
     auto *V = Op->getUnderlyingValue();
     if (!V) {
+      if (VPIntrinsic::isVPIntrinsic(VectorIntrinsicID)) {
+        Arguments.push_back(V);
+        break;
+      }
```

Comment:
Why gather the nullptr into Arguments for vp intrinsic only?
Is it for the all-true mask, or for other reasons?

Reply:
I see, but I don't have better idea for this, since both all-true mask and EVL are not Value when we computing cost. :(
BTW, I think the assertion should be fixed because not all vp intrinsic have mask (vp.select, vp.merge).

Reply:
Please document at least. Also, a bit surprising to just push 1 `nullptr` instead of 2. Are any cost implementations actually using the IR arguments?

Reply:
> I see, but I don't have better idea for this, since both all-true mask and EVL are not Value when we computing cost. :(
So before I took the cost to be based on non-vp intrinsics ~ From BasicTTIImpl.h && RISCVTargetTransform implementation: VP Intrinsics should have the same cost as their non-vp counterparts.  Are we sure use cost is based on vp intrisnic？

Reply:
> Please document at least. Also, a bit surprising to just push 1 `nullptr` instead of 2. Are any cost implementations actually using the IR arguments?

eg: llvm.vp.smx/llvm.vp.smin ...

Reply:
Why you don't push both null mask and null evl?
```suggestion
      if (VPIntrinsic::isVPIntrinsic(VectorIntrinsicID) && 
          (VPIntrinsic::getMaskParamPos(VectorIntrinsicID) == Idx || 
          VPIntrinsic::getVectorLengthParamPos(VectorIntrinsicID) == Idx)) {
        Arguments.push_back(V);
        continue;
      }
```

Reply:
Confirmed, only evl is nullptr currently.
If so, I recommend adding assertion here.
```suggestion
      if (VPIntrinsic::isVPIntrinsic(VectorIntrinsicID)) {
        assert(VPIntrinsic::getVectorLengthParamPos(VectorIntrinsicID) == Idx && 
               "Unexpected null operand of vp intrinsic");
        Arguments.push_back(V);
        break;
      }
```

Reply:
When the CallArguments of IntrinsicRecipe is recipe, it is also nullptr, as follows：
![image](https://github.com/user-attachments/assets/cac4666c-bda6-4a76-bf3b-5c3316749716)